backend/src/services/ReviewService.ts

```diff
@@ -90,23 +90,26 @@ export const createReview = (
     else {
       landlordId = landlord._id;
 
-      // Create property with landlordId just above
-      return createProperty(postal_code, street_name, street_num, landlordId)
-        .then(newProperty => {
-          propertyId = newProperty._id;
+      return getProperty(postal_code, street_name, street_num)
+        .then((foundProperty) => {
+          if (foundProperty) {
+            return foundProperty;
+          } else {
+            // Create property with landlordId just above
+            return createProperty(postal_code, street_name, street_num, landlordId);
+          }
+        })
+        .then(property => {
+          propertyId = property._id;
 
           // ...Finally we have all 2 ids, create Review.
-          return createReviewWithUpdatedValues(title, desc, sentiment, healthSafety, respect, repair, calculatedScore, userId, landlordId)
-            .catch((error) => {
-              console.log(error, "error creating review.");
-              return null;
-            });
+          return createReviewWithUpdatedValues(title, desc, sentiment, healthSafety, respect, repair, calculatedScore, userId, landlordId);
```

Comment:
Check to see if property already exists, if not create a property before creating review